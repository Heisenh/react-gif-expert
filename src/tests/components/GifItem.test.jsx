import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en GifItem', () => {

    const title = 'Killua Zondick';
    const url = 'https://hunterxhunter.com/killua.jpg'

    test('Debe hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con el URL y el alt indicado', () => {

        render( <GifItem title={ title } url={ url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } =  screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    })

    test('Debe de mostrar el titulo en el componente', () => {

        render( <GifItem title={ title } url={ url } />);
        expect( screen.getByText( title )).toBeTruthy();
    })



});