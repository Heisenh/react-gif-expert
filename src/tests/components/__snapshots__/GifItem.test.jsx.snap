// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Killua Zondick"
      src="https://hunterxhunter.com/killua.jpg"
    />
    <p>
      Killua Zondick
    </p>
  </div>
</div>
`;
